### PR TITLE
v2 migration script

### PR DESCRIPTION
**Pull request by pmalek-sumo**
###### Description

Create v2 migration script:

* add migration of `prometheus-operator.prometheusOperator.tlsProxy` -> `kube-prometheus-stack.prometheusOperator.tls`
* rename `prometheus-operator` -> `kube-prometheus-stack`
* rename `prometheus-operator.prometheus.prometheusSpec.containers.(name==prometheus-config-reloader)` -> `config-reloader` as `prometheus-config-reloader` is not supported anymore in prometheus-operator `0.43` [ref](https://github.com/prometheus-operator/prometheus-operator/blob/release-0.43/example/prometheus-operator-crd/monitoring.coreos.com_prometheuses.yaml#L1122-L1132)

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: tests/upgrade_v2_script/static/prometheus_operator_valid.input.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                configMapKeyRef:
+                  name: sumologic-configmap
+                  key: fluentdNamespace
+sumologic:
+  accessId: dummy
+  accessKey: dummy
+  endpoint: http://dummy.com/
+telegraf-operator:
+  enabled: true
+  replicaCount: 1
```
Comment (sumo-drosiek) [inline tests/upgrade_v2_script/static/prometheus_operator_valid.input.yaml:53]:
```suggestion
  replicaCount: 1

```

File: deploy/helm/sumologic/upgrade-2.0.0.sh [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+
+function migrate_pre_upgrade_hook() {
+  # Keep pre-upgrade hook
+  if [[ -n "$(yq r "${TEMP_FILE}" -- sumologic.setup)" ]]; then
+    info "Updating setup hooks (sumologic.setup.*.annotations[helm.sh/hook]) to 'pre-install,pre-upgrade'"
+    yq w -i "${TEMP_FILE}" -- 'sumologic.setup.*.annotations[helm.sh/hook]' 'pre-install,pre-upgrade'
+  fi
+}
+
+function check_falco_state() {
```
Comment (perk-sumo) [inline deploy/helm/sumologic/upgrade-2.0.0.sh:198]: This one is not needed, we don't change the default falce state when updating`1.x -> 2.x`.

File: deploy/helm/sumologic/upgrade-2.0.0.sh [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+  # Print information about falco state
+  if [[ "$(yq r "${TEMP_FILE}" -- falco.enabled)" == 'true' ]]; then
+    info 'falco will be enabled. Change "falco.enabled" to "false" if you want to disable it (default for 2.0)'
+  else
+    info 'falco will be disabled. Change "falco.enabled" to "true" if you want to enable it'
+  fi
+}
```
Comment (perk-sumo) [inline deploy/helm/sumologic/upgrade-2.0.0.sh:205]:
```suggestion
```

File: deploy/helm/sumologic/upgrade-2.0.0.sh [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+check_required_command sed
+
+create_temp_file
+
+migrate_customer_keys
+
+migrate_pre_upgrade_hook
+migrate_prometheus_operator_to_kube_prometheus_stack
+
+check_falco_state
```
Comment (perk-sumo) [inline deploy/helm/sumologic/upgrade-2.0.0.sh:260]: Falco state does not change, no need to check it.
```suggestion
```

**Review by perk-sumo — APPROVED** [review]
Checking `Falco` is unnecessary, otherwise 👍

**Pull request merged**